lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -298,6 +297,31 @@ public void onAuthUpdated() {
 		}
 	}
 
+	/**
+	 * Called when where was an error during authentication attempt
+	 *
+	 * @param errorInfo Error associated with unsuccessful authentication
+	 */
+	public void onAuthError(ErrorInfo errorInfo) {
+		Log.i(TAG, String.format("onAuthError: (%d) %s", errorInfo.code, errorInfo.message));
+		switch (state.state) {
+			case connecting:
+				ITransport transport = this.transport;
+				if (transport != null)
+					/* onTransportUnavailable will send state change event and set transport to null */
+					onTransportUnavailable(transport, null, errorInfo);
```

Comment:
OK, I see that if the transport is closed because of an expired token, and the subsequent attempt to get a new token fails for that reason, this is how we will communicate that failure to the client.

In the case that the transport is still intact, and the server signals that the token is about to expire, and the preemptive token renewal fails: I think in this case we will just log the error instead of setting any error state on the existing connection - and then token will shortly expire and there will be a new attempt. If that second attempt succeeds, we're ok; if it fails, we're in the `connecting` case above, and the client will see that in a connection state change event.

@trenouf does that sound right?